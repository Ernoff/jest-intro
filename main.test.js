/* eslint-dev jest */

// npm packages
const nock = require('nock');
const sinon = require('sinon');
const inquirer = require('inquirer');
// our packages
const {getData, handler} = require('./main');

const gamesCrate = {
  id: '578080',
  name : 'GAMESCOM INVITATIONAL CRATE'
};

// mocks opn function
jest.mock('opn');
const opn = require('opn');

describe('Main function', () => {
  let steamComm;

  beforeAll(() => {

    const url = `/market/priceoverview/?appid=${gamesCrate.id}&country=DE&currenccy=3&market_hash_name=${encodeURIComponent(gamesCrate.name)}`
    steamComm = nock('http://steamcommunity.com')
      .get(url)
      .times(3)
      .reply(200, {
        lowest_price: "$7.25 USD",
        median_price: "$7.19 USD",
        success: true,
        volume: "8,080"
      })
  });

  afterAll(()=> {
    steamComm.cleanAll();
  });

  test('Get default data', async done => {
   const result = await getData(gamesCrate);
    expect(result).toMatchSnapshot();
    done();
  });

  test('Renders correct data to a User', async done => {
    //stub the reply from user
    const inqStub = sinon.stub(inquirer, 'prompt').callsFake(() => ({openBrowser:false}))
    
    // tab in console log function
    const consoleSpy = sinon.spy(console, 'log');
    // execute main logic
    await handler(gamesCrate);
    consoleSpy.restore();
    expect(consoleSpy.args).toMatchSnapshot();
    inqStub.restore();
    done();
  });

  test('Open browser on user request', async done => {
    //stub the reply from user
    const inqStub = sinon.stub(inquirer, 'prompt').callsFake(() => ({openBrowser:true}));
    const consoleSpy = sinon.spy(console, 'log');
    await handler(gamesCrate);
    consoleSpy.restore();
    expect(consoleSpy.args).toMatchSnapshot();
    expect(opn).toBeCalled();
    inqStub.restore();
    done();
  });


});