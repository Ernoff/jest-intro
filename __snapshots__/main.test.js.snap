// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main function Get default data 1`] = `
Object {
  "lowest_price": "$7.25 USD",
  "median_price": "$7.19 USD",
  "success": true,
  "volume": "8,080",
}
`;

exports[`Main function Open browser on user request 1`] = `
Array [
  Array [
    "Right Now [1mGamescom Invitational Crate Costs[22m: 
    [31m- Lowest price:[39m  [32m$7.25 USD[39m 
    [31m- Median price:[39m  [34m$7.19 USD[39m
    [31m- Volume:[39m        [33m8,080[39m",
  ],
]
`;

exports[`Main function Renders correct data to a User 1`] = `
Array [
  Array [
    "Right Now [1mGamescom Invitational Crate Costs[22m: 
    [31m- Lowest price:[39m  [32m$7.25 USD[39m 
    [31m- Median price:[39m  [34m$7.19 USD[39m
    [31m- Volume:[39m        [33m8,080[39m",
  ],
]
`;
